test/test.js

```diff
@@ -44,6 +85,59 @@ describe('vhost(hostname, server)', function () {
     .expect(200, 'loopback', done)
   })
 
+  it('should support IPv6 literal in Host with no port', function (done) {
+    var app = createServer('[::1]', function (req, res) {
+      res.end('loopback')
+    })
+
+    request(app)
+    .get('/')
+    .set('Host', '::1')
```

Comment:
My bad, from briefly reading the ipv6 wiki page i falsely assumed that they didnt need to have square brackets if there was no port.